hazelcast/src/test/java/com/hazelcast/jet/config/JobConfigTest.java

```diff
@@ -231,50 +225,30 @@ public void addCustomClasspaths() {
         );
     }
 
+    /**
+     * Assert that any mutating method (defined as any {@link JobConfig} method returning itself, a fluent configuration) throws
+     * a {@link IllegalStateException} after {@link JobConfig#lock} has been called
+     */
     @Test
-    @SuppressWarnings("rawtypes")
     public void when_mutatingLockedJobConfig_then_fail() {
-        JobConfig jobConfig = new JobConfig();
+        final JobConfig jobConfig = new JobConfig();
 
-        URL mockUrl;
-        try {
-            File file = File.createTempFile("jobConfig", "suffix");
-            mockUrl = file.toURI().toURL();
-        } catch (Exception e) {
-            throw new RuntimeException(e);
-        }
+        jobConfig.lock();
 
-        List<Supplier> mutatingMethods = Arrays.asList(
-                () -> jobConfig.setName(""),
-                () -> jobConfig.setSplitBrainProtection(false),
-                () -> jobConfig.setAutoScaling(false),
-                () -> jobConfig.setSuspendOnFailure(false),
-                () -> jobConfig.setProcessingGuarantee(null),
-                () -> jobConfig.setSnapshotIntervalMillis(0L),
-                () -> jobConfig.addClass(this.getClass()),
-                () -> jobConfig.addPackage(this.getClass().getPackage().getName()),
-                () -> jobConfig.addJar(mockUrl),
-                () -> jobConfig.addJarsInZip(mockUrl),
-                () -> jobConfig.addClasspathResource(mockUrl),
-                () -> jobConfig.addCustomClasspath(null, null),
-                () -> jobConfig.addCustomClasspaths(null, null),
-                () -> jobConfig.attachFile(mockUrl),
-                () -> jobConfig.attachDirectory(""),
-                () -> jobConfig.attachAll(null),
-                () -> jobConfig.registerSerializer(null, null),
-                () -> jobConfig.setArgument("", ""),
-                () -> jobConfig.setClassLoaderFactory(null),
-                () -> jobConfig.setInitialSnapshotName(""),
-                () -> jobConfig.setMetricsEnabled(false),
-                () -> jobConfig.setStoreMetricsAfterJobCompletion(false),
-                () -> jobConfig.setMaxProcessorAccumulatedRecords(0L),
-                () -> jobConfig.setTimeoutMillis(0L)
-        );
+        final Stream<Method> mutatorMethods = Arrays.stream(jobConfig.getClass().getMethods())
+                .filter(method -> method.getReturnType().equals(jobConfig.getClass()));
 
-        jobConfig.lock();
-        for (Supplier mutatingMethod : mutatingMethods) {
-            assertThrows(IllegalStateException.class, mutatingMethod::get);
-        }
+        mutatorMethods.forEach(method -> {
+            // Set parameters:
+            // Reference types to null,
```

Comment:
passing null to `@Nonnull` parameter is not the best idea. You rely on order of checks (locked vs argument validation)